persistent/Database/Persist.hs

```diff
@@ -6,7 +6,30 @@ module Database.Persist
     ( module Database.Persist.Class
     , module Database.Persist.Types
 
-      -- * Query update combinators
+    -- * #dataset# Reference Schema & Dataset
+    -- |
+    --
+    -- All the combinators present here will be explained based on this schema:
+    --
+    -- > share [mkPersist sqlSettings, mkMigrate "migrateAll"] [persistLowerCase|
+    -- > User
+    -- >     name String
+    -- >     age Int
+    -- >     deriving Show
+    -- > |]
+    --
+    -- and this dataset. From here on we will call this as dataset-1
```

Comment:
> From here on we will call this as dataset-1

"call this as" doesn't sound correct, it'd be better to drop the "as". I think something more explicit would be even better:  "The examples below will refer to this as dataset-1"